src/UserSpaceInstrumentation/TestProcess.cpp

```diff
@@ -0,0 +1,94 @@
+// Copyright (c) 2021 The Orbit Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+#include "TestProcess.h"
+
+#include <absl/time/clock.h>
+
+#include <chrono>
+#include <cstdio>
+#include <fstream>
+#include <string>
+#include <vector>
+
+#include "OrbitBase/Logging.h"
+
+namespace orbit_user_space_instrumentation {
+
+namespace fs = std::filesystem;
+
+namespace {
+
+// Create a file at path.
+void Touch(fs::path p) {
+  std::ofstream ofs(p);
+  ofs << "\n";
+}
+
+}  // namespace
+
+TestProcess::TestProcess() {
+  flag_file_run_child_ = std::tmpnam(nullptr);
+  flag_file_child_started_ = std::tmpnam(nullptr);
+  Touch(flag_file_run_child_);
+  pid_ = fork();
+  CHECK(pid_ != -1);
+  // Start the workload and have the parent wait for the startup to complete.
+  if (pid_ == 0) {
+    DummyWorkload();
+    exit(0);
+  } else {
+    while (!fs::exists(flag_file_child_started_)) continue;
+  }
+}
+
+TestProcess::~TestProcess() {
+  fs::remove(flag_file_run_child_);
+  int status;
+  waitpid(pid_, &status, 0);
+  CHECK(WIFEXITED(status));
+  fs::remove(flag_file_child_started_);
+}
+
+void TestProcess::DummyWorker() {
+  constexpr auto kTimeToLiveMs = std::chrono::milliseconds(15);
+  const auto deadline = std::chrono::system_clock::now() + kTimeToLiveMs;
+  while (true) {
+    if (std::chrono::system_clock::now() > deadline) {
+      break;
+    }
+  }
+  absl::MutexLock lock{&joinable_threads_mutex_};
+  joinable_threads_.emplace(std::this_thread::get_id());
+}
+
+void TestProcess::DummyWorkload() {
```

Comment:
Fake?

Reply:
It isn't a fake (in my understanding of fake). Just dropped "dummy".